src/content/app/genome-browser/hooks/useFocusTrack.ts

```diff
@@ -77,22 +85,58 @@ const useFocusGene = (params: Params) => {
   const focusObjectIdRef = useRef(focusObjectId);
   const geneIdRef = useRef(geneStableId);
   const visibleTranscriptIds = focusGene?.visibleTranscriptIds ?? null;
-  const trackSettingsForGenome = useAppSelector(getAllTrackSettings)
+  const focusGeneTrackSettings = useAppSelector(getAllTrackSettings)
     ?.settingsForIndividualTracks.focus as FocusGeneTrack | undefined;
 
+  const allSortedFocusGeneTranscriptsRef = useRef<string[]>([]);
+  const visibleTranscriptIdsRef = useRef(visibleTranscriptIds);
+  const showSeveralTranscriptsRef = useRef(
```

Comment:
Curious why refs are used here in favour of state

Reply:
- Setting a state causes a rerender
- All refs in this hook are intended to overcome stale closures (e.g. you subscribe to genome browser events early in the lifetime of the component; but you have to use up-to-date variables inside of the subscriber functions). Refs, being mutable, help with that. State, being immutable, doesn't.

Reply:
That makes sense. Thanks.